File: backend/app/plugins/simulators/press_feed/manifest.py
from app.plugins.base import PluginCategory, PluginManifest

manifest = PluginManifest(
    code="press_feed",
    name="Fil presse",
    name_en="Press Feed",
    description="Simulation de flux actualites media",
    category=PluginCategory.SIMULATOR,
    icon="Newspaper",
    default_color="gray",
    supported_formats=["text", "image"],
    default_enabled=False,
    coming_soon=False,
    sort_order=6,
    inject_types=[],
    router_module="app.routers.simulated_channels",
    router_prefix="/api/simulated",
)

File: frontend/src/plugins/simulators/press_feed/manifest.ts
import { lazy } from 'react'
import { Newspaper } from 'lucide-react'
import type { PluginManifest } from '../../types'

export const manifest: PluginManifest = {
  code: 'press_feed',
  name: 'plugins.press_feed.name',
  description: 'plugins.press_feed.description',
  category: 'simulator',
  icon: Newspaper,
  defaultColor: 'gray',
  supportedFormats: ['text', 'image'],
  defaultEnabled: false,
  comingSoon: false,
  sortOrder: 6,
  injectTypes: [],
  playerRoute: '/press',
  PlayerPage: lazy(() => import('./PlayerPage')),
  EventReceiver: lazy(() => import('./EventReceiver')),
  PreviewPopup: lazy(() => import('./PreviewPopup')),
  configSchema: null,
}
